### PR TITLE
Not add a cert to CA cache if it doesn't set "CA:TRUE" as basic constraints

### PR DESCRIPTION
**Pull request by miyazakh**
# Description

Not add a cert to CA cache if it doesn't have `CA:TRUE` as basic constraints. The behavior is enabled  when OPENSSL_ALL is defined. This change is needed for qt nightly Jenkins test failure fix.

Fix trusted peer cert cache
 It could not add a cert to trusted peer cert cache if the cert has the same subject as pre-added cert. For example, `./certs/server-ecc-self.pem`
 The cert above has the same subject to server-ecc.pem. Therefore, it could not add the cert to cache if there is "server-ecc.pem" in trusted peer cert cache already. This was revealed after changing "Not add a cert to CA cache".

# Testing

Qt jenkins test. Unit test

# Checklist

 - [X] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by miyazakh** [comment]
retest this please

File: certs/intermediate/ca_false_intermediate/gentestcert.sh [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    sed -i "s/_CA_NAME_/$INTCA_NAME/" "$OPENSSL_INTCA_CONF"
+    sed -i "s/_CERT_NAME_/$SRVCERT_NAME/" "$OPENSSL_INTCA_CONF"
+    sed -i "s/_CA_DEPART_/Product_Support/" "$OPENSSL_INTCA_CONF"
+}
+
+cleanup_files(){
+    rm -f wolfssl_ca.conf
+    rm -f wolfssl_int_ca.conf
+    rm -rf pki/
+    exit 0
```
Comment (douzzer) [inline certs/intermediate/ca_false_intermediate/gentestcert.sh:78]: the `exit 0` here is masking the `echo "Completed"` at the end of the script.

**Review by douzzer — CHANGES_REQUESTED** [review]
Found a few nits in `certs/intermediate/ca_false_intermediate/gentestcert.sh`.

With `--enable-all CFLAGS='-DWOLFSSL_X509_STRICT'`, I'm seeing the follow unit test failures (rebase on 3e1f365e75 to assure line numbers match):
```
   686: test_wolfSSL_OCSP_REQ_CTX                           :
ERROR - tests/api.c line 76037 failed with:
    expected: wolfSSL_X509_LOOKUP_load_file(lookup, "certs/ocsp/server1-cert.pem", WOLFSSL_FILETYPE_PEM) == 1
    result:   0 != 1

[...]

   793: test_wolfSSL_CertManagerLoadCABuffer_ex             :
ERROR - tests/api.c line 3037 failed with:
    expected: ret == (ASN_AFTER_DATE_E)
    result:   -357 != -151

[...]

   848: test_wolfSSL_CTX_load_verify_locations              :
ERROR - tests/api.c line 2573 failed with:
    expected: wolfSSL_CTX_load_verify_locations_ex(ctx, ((void *)0), load_expired_path, 0x00000002 | 0x00000004) == WOLFSSL_SUCCESS
    result:   0 != 1
```

File: certs/intermediate/ca_false_intermediate/gentestcert.sh [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+mv ./pki/$CA_NAME/$CA_NAME.pem $CA_PEM
+
+# Intermediate CA
+openssl x509 -in $INT_CA_HOME/$INTCA_NAME.crt -inform PEM -noout -text > $INT_CA_HOME/$INTCA_NAME.pem
+cat $INT_CA_HOME/$INTCA_NAME.crt >> $INT_CA_HOME/$INTCA_NAME.pem
+mv $INT_CA_HOME/$INTCA_NAME.pem $INTCA_PEM
+# Server
+openssl x509 -in $SRV_CRT_HOME/server.crt -inform PEM -noout -text > $SRV_CRT_HOME/server.pem
+cat $SRV_CRT_HOME/server.crt >> $SRV_CRT_HOME/server.pem
+mv $SRV_CRT_HOME/server.pem $SEVER_PEM
```
[the remark this answers is not in the document]
Comment (miyazakh) [reply certs/intermediate/ca_false_intermediate/gentestcert.sh]: Thanks

**Comment by miyazakh** [comment]
Fixed unit test failures with `--enable-all CFLAGS='-DWOLFSSL_X509_STRICT'`

File: certs/intermediate/ca_false_intermediate/gentestcert.sh (hunk 1 above)
Comment (miyazakh) [reply certs/intermediate/ca_false_intermediate/gentestcert.sh]: moved "exit"

**Comment by cconlon** [comment]
@miyazakh Please fix merge conflicts, then re-assign to @douzzer / @wolfSSL-Bot, thanks.

**Comment by miyazakh** [comment]
Re-visited this PR based on [PR8087](https://github.com/wolfSSL/wolfssl/pull/8087). 
Removed WOLFSSL_X509_STRICT macro and WOLFSSL_MUST_BE_CA enum

**Comment by miyazakh** [comment]
Thanks!

**Comment by dgarske** [comment]
Retest this please. History for PRB lost

**Review by douzzer — APPROVED** [review]

**Pull request merged**